tests/test-edit-flow-notifications.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+class WP_Test_Edit_Flow_Notifications extends WP_UnitTestCase {
+
+	protected static $admin_user_id;
+	protected static $EF_Notifications;
+		
+	public static function wpSetUpBeforeClass( $factory ) {
+		global $edit_flow;
+
+		/**
+		 * `install` is hooked to `admin_init` and `init` is hooked to `init`.
+		 * This means when running these tests, you can encounter a situation
+		 * where the custom post type taxonomy has not been loaded into the database
+		 * since the tests don't trigger `admin_init` and the `install` function is where
+		 * the custom post type taxonomy is loaded into the DB. 
+		 * 
+		 * So make sure we do one cycle of `install` followed by `init` to ensure 
+		 * custom post type taxonomy has been loaded.
+		 */
+		$edit_flow->custom_status->install();
+		$edit_flow->custom_status->init();
+
+		self::$admin_user_id = $factory->user->create( array( 'role' => 'administrator' ) );
```

Comment:
In general, probably better to create the user on `setUp` so it's reset between tests?

Reply:
From [poking around](https://github.com/Automattic/jetpack/blob/384678b75d1ccdbb3f8bdd23bff21c9d6045a4fb/tests/php/core-api/wpcom-fields/test-class.subscribers-endpoint.php#L15) it seemed like most folks were creating users in `wpSetUpBeforeClass` and in individual tests (would `$factory` be available in `setUp`?)

Reply:
> would $factory be available in setUp?

Yep, it would.

I find it's generally safer and better to set up the test data on a per-test basis (wherever possible). It's a bit slower but not as risky (bugs in tests are super painful to debug).

I don't think it's a big deal here since we're not manipulating the user in any way, so I'd be okay with leaving it.

Reply:
Ah interesting (https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#shared-fixtures):

> Sometimes, all tests in a class can use the same database fixtures. Generating these fixtures inside of setUp() will cause them to be recreated for each test, which can dramatically decrease test performance. In these cases, it’s possible to create fixtures once, before any tests in the class have run, using wpSetUpBeforeClass()

Reply:
Ah hey, good to know